bulbs/contributions/models.py

```diff
@@ -1,19 +1,127 @@
+        for contribution in qs.all():
+            rate = contribution.get_rate()
+            if rate and hasattr(rate, "rate"):
+                pay += rate.rate
+        return pay
+
+
+class LineItem(models.Model):
+    contributor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    amount = models.IntegerField(default=0)
+    note = models.TextField()
+    payment_date = models.DateTimeField(null=True, blank=True)
+
 
 class ContributorRole(models.Model):
     name = models.CharField(max_length=255)
     description = models.TextField(null=True, blank=True)
+    payment_type = models.IntegerField(choices=ROLE_PAYMENT_TYPES, default=MANUAL)
 
 
 class Contribution(models.Model):
     role = models.ForeignKey(ContributorRole)
-    contributor = models.ForeignKey(settings.AUTH_USER_MODEL)
-    content = models.ForeignKey(Content)
+    contributor = models.ForeignKey(settings.AUTH_USER_MODEL, related_name="contributions")
+    content = models.ForeignKey(Content, related_name="contributions")
     notes = models.TextField(null=True, blank=True)
+    minutes_worked = models.IntegerField(null=True)
+    force_payment = models.BooleanField(default=False)
+    payment_date = models.DateTimeField(null=True, blank=True)
+
+    def get_rate(self):
+        payment_type = self.role.payment_type
+        if payment_type == MANUAL:
+            return self.manual_rates.all().first()
+
+        if payment_type == FLAT_RATE:
+            return self.role.flat_rates.filter().first()
+
+        if self.content.feature_type and payment_type == FEATURETYPE:
+            return self.content.feature_type.feature_type_rates.all().first()
+
+        if payment_type == HOURLY:
+            return self.role.hourly_rates.filter().first()
+
+
+class Rate(models.Model):
+    name = models.IntegerField(choices=RATE_PAYMENT_TYPES, null=True)
+    updated_on = models.DateTimeField(auto_now=True)
+    rate = models.IntegerField()
+
+    class Meta:
+        ordering = ('-updated_on',)
+
+
+class FlatRate(Rate):
+    role = models.ForeignKey(ContributorRole, related_name="flat_rates")
+
+
+class HourlyRate(Rate):
+    role = models.ForeignKey(ContributorRole, related_name="hourly_rates")
+
+
+class ManualRate(Rate):
+    contribution = models.ForeignKey(Contribution, related_name="manual_rates")
+
+
+class FeatureTypeRate(Rate):
+    role = models.ForeignKey(ContributorRole, null=True, related_name="feature_type_rates")
+    feature_type = models.ForeignKey(FeatureType, related_name="feature_type_rates")
+
+
+class Override(PolymorphicModel):
+    name = models.IntegerField(choices=RATE_PAYMENT_TYPES, null=True)
+    updated_on = models.DateTimeField(auto_now=True)
+    rate = models.IntegerField()
+    contributor = models.ForeignKey(
+        settings.AUTH_USER_MODEL, related_name="overrides", null=True
+    )
+    role = models.ForeignKey(ContributorRole, related_name="overrides", null=True)
+
+    objects = PolymorphicManager()
+
+
+class ContributionOverride(Override):
```

Comment:
What is this used for, and does it need a custom serializer like FeatureTypeOverride?